src/pages/Login.js

```diff
@@ -1,7 +1,61 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { saveTokens, saveUser } from '../services/storage';
 
-const Login = () => (
-  <div>login</div>
-);
+const Login = () => {
+  const [disabled, setDisabled] = useState(true);
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const history = useHistory();
+  useEffect(() => {
+    const validateEmail = (value) => {
+      const re = /\S+@\S+\.\S+/;
+      return re.test(value);
+    };
+    const passwordMin = 6;
+    if (validateEmail(email) && password.length > passwordMin) {
+      setDisabled(false);
+    } else {
+      setDisabled(true);
+    }
```

Comment:
Oi, gente!
Gostei do código, mas gostaria de fazer uma sugestão mais "legível" que é a seguinte:

```suggestion
    const emailIsValid =  /\S+@\S+\.\S+/.test(email);
    const passwordMin = 6;
    const passwordIsValid = password.length > passwordMin;
    const buttonIsDisabled = !(emailIsValid && passwordIsValid);
    setDisabled(buttonIsDisabled);
```
Se gostarem, é só aceitar o commit.